### PR TITLE
Equal Sum Partition Problem using Dynamic Programming in C++

### PR DESCRIPTION
**Pull request by nandinit02**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
Issue: #2399 
I have added the code for equal sum partition problem using dynamic programming approach in C++, using proper explanation of code, sample test cases, time complexity and space complexity of the code.

### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue: #2399

### CODE REVIEW

**Comment by nandinit02** [comment]
Hey, @HarshCasper @ankitaggarwal23 , please do review my pr
Thank you.

File: C-Plus-Plus/dp/.cph/.Equal_sum_partition.cpp_d130310b148dc436cb9052663ab3183a.prob [hunk 1]
```diff
@@ -0,0 +1 @@
+{"name":"Local: Equal_sum_partition","url":"c:\\Users\\Nandini Tiwari\\Desktop\\Git-Demo\\NeoAlgo\\C-Plus-Plus\\dp\\Equal_sum_partition.cpp","tests":[{"id":1616080413179,"input":"","output":""}],"interactive":false,"memoryLimit":1024,"timeLimit":3000,"srcPath":"c:\\Users\\Nandini Tiwari\\Desktop\\Git-Demo\\NeoAlgo\\C-Plus-Plus\\dp\\Equal_sum_partition.cpp","group":"local","local":true}
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/.cph/.Equal_sum_partition.cpp_d130310b148dc436cb9052663ab3183a.prob]: Remove this

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/dp/.cph/.Equal_sum_partition.cpp_d130310b148dc436cb9052663ab3183a.prob (hunk 1 above)
Comment (nandinit02) [reply C-Plus-Plus/dp/.cph/.Equal_sum_partition.cpp_d130310b148dc436cb9052663ab3183a.prob]: Hey @HarshCasper , I have made the required changes please check.

**Comment by nandinit02** [comment]
Hi @HarshCasper, I have made the required changes, please review my pr.
Thankyou.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**